src/runtime-rs/crates/hypervisor/ch-config/src/convert.rs

```diff
@@ -481,10 +599,29 @@ where
     Ok(MacAddr { bytes })
 }
 
+// Determine if the specified Share Fs (virtiofs mount) can be safely ignored.
+//
+// Normally, the container rootfs is passed as a virtiofs device.
+// However, if the user requested the container rootfs be passed
+// as a block device (for example if using containerd with the
+// devmapper snapshotter), runtime-rs will pass the container
+// rootfs _twice_: once as a virtiofs device and once as a block
+// device. Since TDX does not permit the use of virtiofs, ignore
+// the virtiofs request as we'll actually use the block device
+// request instead.
+//
+// See: https://github.com/kata-containers/kata-containers/issues/7125
+//
+pub fn can_ignore_share_fs(mount_tag: &str, guest_protection_to_use: GuestProtection) -> bool {
+    mount_tag == MOUNT_GUEST_TAG && guest_protection_is_tdx(guest_protection_to_use)
```

Comment:
I am good with using the `guest_protection_is_tdx` check for now. But shouldnt this be more of a devicemapper check like `is_storge_driver_devicemapper` to make sure this works with devicemapper as well without the confidential case .
If so, maybe track this as well as an issue?